fix(bazel): specify the package and lock files using the workspace

alan-agius4 opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?
When used downstream (CLI), The following error is being thrown
```bash
no such package '@ts-api-guardian_runtime_deps//': Unable to load package for //tools/ts-api-guardian:package.json: not found. and referenced by '//etc/api:angular_devkit_schematics_tasks_api'

no such package 'tools/ts-api-guardian': BUILD file not found on package path and referenced by '//etc/api:angular_devkit_schematics_testing_api.accept'
```

<!-- Please check the one that applies to this PR using "x". -->
```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

//cc @alexeagle

alan-agius4 committed:
fix(bazel): specify the package and lock files using the workspace

alexeagle commented on tools/ts-api-guardian/index.bzl:
```diff
@@ -17,14 +17,12 @@
 
 load("@build_bazel_rules_nodejs//internal/node:node.bzl", "nodejs_binary", "nodejs_test")
 
-COMMON_MODULE_IDENTIFIERS = ["angular", "jasmine", "protractor"]
```
I'd rather keep this constant (it's technically part of the public API for this rule)
and use it below as the default for the named param

alexeagle replied:
I went ahead and made the change so we can get this greened up

alan-agius4 commented:
Thanks @alexeagle!

On Fri, 07 Sep 2018 at 16:38, Alex Eagle <notifications@github.com> wrote:

> *@alexeagle* commented on this pull request.
> ------------------------------
>
> In tools/ts-api-guardian/index.bzl
> <https://github.com/angular/angular/pull/25694#discussion_r215980527>:
>
> > @@ -17,14 +17,12 @@
>
>  load("@build_bazel_rules_nodejs//internal/node:node.bzl", "nodejs_binary", "nodejs_test")
>
> -COMMON_MODULE_IDENTIFIERS = ["angular", "jasmine", "protractor"]
>
> I went ahead and made the change so we can get this greened up
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular/pull/25694#discussion_r215980527>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AQv-Wq_EFhrs2sX7CZtIGZ-ldxzbL5nFks5uYoT9gaJpZM4WO8Km>
> .
>

alan-agius4 committed:
feat(bazel): add additional parameters to `ts_api_guardian_test` def

Added `strip_export_pattern` and `allow_module_identifiers` so that these can be passed from downstream

IgorMinar commented:
This PR doesn't apply cleanly to the patch branch. @alan-agius4 can you send a separate PR for that branch please?

alan-agius4 commented:
Yeah I can do that tomorrow. Most likely it will have to contain the
changes of the following PR as well
https://github.com/angular/angular/pull/25690 as it was merged only to
master.




On Fri, 07 Sep 2018 at 22:51, Igor Minar <notifications@github.com> wrote:

> This PR doesn't apply cleanly to the patch branch. @alan-agius4
> <https://github.com/alan-agius4> can you send a separate PR for that
> branch please?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular/pull/25694#issuecomment-419561710>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQv-WhQnsu23ptaRimRdxHrDxyGQPwjjks5uYtxjgaJpZM4WO8Km>
> .
>
